__tests__/endpoint.test.js

```diff
@@ -102,3 +102,92 @@ describe("/api/articles", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: adds a comment to an article", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "This is a test comment.",
+    };
+    const articleId = 1;
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const addedComment = response.body;
+        expect(addedComment).toHaveProperty("comment_id");
+        expect(addedComment).toHaveProperty("author", newComment.username);
+        expect(addedComment).toHaveProperty("body", newComment.body);
+        expect(addedComment).toHaveProperty("article_id", 1);
+        expect(addedComment).toHaveProperty("created_at");
+        expect(addedComment).toHaveProperty("votes", 0);
+      });
+  });
+  test("201: ignores extra property", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "This is a test comment.",
+      foo: "Unnecessary Property",
+    };
+    const articleId = 1;
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const addedComment = response.body;
+        expect(addedComment).not.toHaveProperty("foo");
+      });
+  });
+  test("400: wrong data type", () => {
+    const articleId = "hello";
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({ msg: "Bad Request" });
+      });
+  });
+  test("400: missing required fields", () => {
+    const newComment = {
+      username: "",
+      body: "",
+    };
```

Comment:
Almost there. Im happy for you to keep this test as it will act as a backstop to people who try to post empty comment bodies 👍 

However, we'd like a similar test where one of the keys is missing entirely.